### ISSUE TITLE
Responses reference not followed

### ISSUE THREAD

**Issue by kierans** [outofcoffee/imposter#39]
Firstly thankyou for this tool. I didn't think it would be so hard to find a tool that would mock/stub the spec and return example responses.

However the `ExampleService` [assumes that the response is inlined](https://github.com/outofcoffee/imposter/blob/master/plugin/openapi/src/main/java/io/gatehill/imposter/plugin/openapi/service/ExampleServiceImpl.java#L54) ie:
```javascript
{
  "responses": {
    "200": {
      "description": "Successful response",
      "content": {
        "application/json": {
          "schema": {
            ...
          },
          "examples": {
            "application/json": {
              "value": {
                ...               
              }
            }
          }
        }
      }
    }
  }
}
```

If however, I use a [Reference Object](https://swagger.io/specification/#responsesObject) the example is not found ie:
```javascript
{
  "responses": {
    "200": {
      "$ref": "#/components/responses/myResponse"
    }
  }
}
```

Unfortunately my entire spec uses response references, so for me to use this tool I need the reference to be followed.

**Comment by outofcoffee** [comment]
Hi @kierans, thank you very much for raising this. This shouldn't be too tough to implement, as it's a variation of something we do already. Would you be able to provide a full specification to make sure we can cover your use case?

**Comment by kierans** [comment]
I've taken the [Petstore example](https://github.com/OAI/OpenAPI-Specification/blob/master/examples/v3.0/petstore.yaml) and modified it. Please see attached [petstore.yml.txt](https://github.com/outofcoffee/imposter/files/4534689/petstore.yml.txt)

If the response is inlined the array of pets is returned, but as a Response Reference they are not.

**Comment by outofcoffee** [comment]
Great, thanks @kierans. The https://github.com/outofcoffee/imposter/commits/feature/response-refs branch has the changes, with a passing test. Still WIP, as will need some tidy up.

**Comment by outofcoffee** [comment]
#40 is now merged and should be available as the `beta` tag for all Docker images shortly.

**Comment by kierans** [comment]
@outofcoffee I've tested the `beta` build (`320a490ba9a4`) and my examples are returned with Response Refs. 👍 

When are you planning on making the next release?

**Comment by outofcoffee** [comment]
Thanks @kierans. I've released [v1.2.0](https://github.com/outofcoffee/imposter/releases/tag/v1.2.0) with this :)
